Dip/Dip/Dip.swift

```diff
@@ -42,8 +42,7 @@ public class DependencyContainer {
     case Int(IntegerLiteralType)
   }
   
-  private var dependencies = [DefinitionKey : Definition]()
-  private var lock: OSSpinLock = OS_SPINLOCK_INIT
```

Comment:
My initial point was that it less error prone to ensure single thread access to shared resource (container in our case) by client then to make container itself guaranty thread safety in all possible cases. Just document that `resolve` should be used from single thread (probably the main?). Really I think it will be very rare case when user will call `resolve` from somewhere else then main thread.